src/components/Dashboard.tsx

```diff
@@ -69,12 +70,12 @@ const TemplateComponent: React.FC<{ template: TemplateInList }> = ({ template })
               {template?.namePlural || `${name} ${strings.LABEL_APPLICATIONS}`}
             </a>
             <Icon name="chevron right" />
-            {loadingFilters && 
+            {loadingFilters !== filters.length &&
               <LoadingSmall/>
             }
           </Label>
           {filters.map((filter, i:number, filters) => (
-            <FilterComponent key={filter.id} template={template} filter={filter} setLoading={setLoadingFilters} filters={filters} i={i}/>
+            <FilterComponent key={filter.id} template={template} filter={filter} setLoadingFilters={setLoadingFilters} filters={filters} loadingFilters={loadingFilters}/>
```

Comment:
Also -- I don't think you've got your Prettier auto-formatting configured correctly. Cos once this auto-formats, it should look like:
```ts
         {filters.map((filter, i: number, filters) => (
            <FilterComponent
              key={filter.id}
              template={template}
              filter={filter}
              setLoadingFilters={setLoadingFilters}
              filters={filters}
            />
          ))}
```

Let me know if you can't get it working -- I suggest having "Format on Save" enabled then you never need to think about it.